src/Char.hpp

```diff
@@ -38,7 +38,7 @@ struct Char {
     Descriptor *desc{};
     AffectList affected;
     NOTE_DATA *pnote{};
-    OBJ_DATA *carrying{};
+    GenericList<OBJ_DATA *> carrying;
```

Comment:
I noticed AffectList contains a GenericList of AFFECT_DATA,  while this one is a list of OBJ_DATA pointers.  And I'm wondering what the pattern ought to be going forward.

Reply:
Right! This is owning, vs-non-owning.

The AffectList _owns_ the objects. They're part of the list, and when the list is destroyed so are the AFFECT_DATAs inside.

The carrying is a _reference_ (in the non-c++ sense) to the carried object. In this case the owner is (sort of) the global `object_list`. At least, that's the thing responsible for tidying it up.

Right now the code hasn't been refactored to use C++ trickery to make the "object_list" actually "own" the data; `extract_obj` is responsible for removing it from all the lists and then ultimately `delete`-ing it.

I went back and forth on this a number of times. Trying to retrofit strict C++ style ownership is the goal but it was already a big change so I left `extract_obj` et al alone. `AffectList` had already been refactored to "own" its items so I changed that.

In the longer-ish term I can actually see us moving to a model where 99% of `extract_obj` goes into the OBJ_DATA destructor, and then extract_obj then becomes "remove this object from the object_list" (which would then "own" the OBJ_DATA either by it being a `GenericList<OBJ_DATA>` or something like `GenericList<unique_ptr<OBJ_DATA>>`.

I really this is really very discursive: I have been back and forth on this a dozen times: most of the time I spent on this whole PR was analysis paralysis on these kinds of decisions; so for now I've chosen a local minima to get something in that's an incremental improvement (I hope!)

Reply:
makes sense